R/drop_search.R

```diff
@@ -1,46 +1,57 @@
 
 
+
 #'Returns metadata for all files and folders whose filename contains the given
 #'search string as a substring.
 #'
 #'@param query  The search string. This string is split (on spaces) into
 #'  individual words. Files and folders will be returned if they contain all
 #'  words in the search string.
 #'@template path
-#'@template file_limit
-#'@param  include_deleted If this parameter is set to true, then files and
-#'  folders that have been deleted will also be included in the search.
-#'@template locale
-#'@template include_membership
+#'@param start The starting index within the search results (used for paging).
+#'  The default for this field is 0
+#'@param max_results The maximum number of search results to return. The default
+#'  for this field is 100.
+#'@param  mode Mode can take the option of filename, filename_and_content, or search deleted files with deleted_filename
 #'@template token
 #'@export
 #' @examples \dontrun{
 #' # If you know me, you know why this query exists
 #' drop_search('gif') %>% select(path, is_dir, mime_type)
 #'}
 drop_search <- function(query = NULL,
```

Comment:
leaving query as NULL gives 400 error; might make more sense to drop the default and let R handle required-argument-not-given scenario

Reply:
I added an assertive assert is not null to make sure a query is passed. Without a query, there is nothing to search for.

Reply:
Fixed in `8b05b88`

Reply:
why not remove `= NULL` altogether? changes error from

```
Error: is_not_null : query is NULL.
```
to
```
Error in drop_search() : argument "query" is missing, with no default
```

where I'd think the latter is the expected R convention.

Reply:
ok, that makes sense. I pushed an update and also added a couple of other checks (to make sure `start` and `max_results` are not negative)